### PR TITLE
add configuration for STS parallel message processing

### PR DESCRIPTION
**Pull request by alanrenmsft**
Add configuration to control the STS message processing behavior.

image

PR for STS: https://github.com/microsoft/sqltoolsservice/pull/1482

STS changes since current version:
image

### CODE REVIEW

**Comment by alanrenmsft** [comment]
@Charles-Gagnon are you aware of the issue with the PR validation build? access token expiration?

File: extensions/mssql/package.json [hunk 1]
```diff
@@ -383,6 +383,11 @@
           "type": "boolean",
           "description": "%mssql.ignorePlatformWarning%",
           "default": false
+        },
+        "mssql.parallelMessageProcessing": {
+          "type": "boolean",
+          "description": "%mssql.parallelMessageProcessing%",
+          "default": false
```
Comment (Charles-Gagnon) [inline extensions/mssql/package.json:390]: I think it'd be a good idea to : 

1. Force this to true for dev builds (easy enough to revert if we start hitting issues)
2. Wait a week or so and if there's no major issues default to true on insiders

It'll be important to get people testing this as much as possible - and if we just leave it as default disabled then I imagine we're not really going to get much actual testing done.

File: extensions/mssql/src/utils.ts [hunk 2]
```diff
@@ -92,6 +92,15 @@ export function getConfigTracingLevel(): string {
 	}
 }
 
+export function getConfigParallelMessageProcessing(): boolean {
+	let config = getConfiguration();
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/utils.ts:96]: `return getConfiguration()?.['parallelMessageProcessing']`

File: extensions/mssql/package.json (hunk 1 above)
Comment (alanrenmsft) [reply extensions/mssql/package.json:390]: that sounds good to me, I was planning to ask folks to turn it on, i guess have it on for dev build by default is a better idea :)

Comment (alanrenmsft) [reply extensions/mssql/package.json:390]: having per quality level default value is actually not as straightforward as I was expecting. in ADS core, the configuration default is in code file, we can check the quality there, but this one is in package.json. we have to rely on the mixing step to change the value. 

are you aware of easier approaches? if no, I'd probably go with the original plan.

Comment (Charles-Gagnon) [reply extensions/mssql/package.json:390]: Yeah, I wish there was a way to make it better. The only solution I can come up with is : 

1. Have a config flag `setDefaultParallelMessageProcessingValue` that you check when extension starts up
2. If it doesn't exist or is true and quality is insiders then update the config value and set `setDefaultParallelMessageProcessingValue` to false

That way it only does it once - so if the user then changes the value themselves it keeps that new value forever (unless they wipe our their settings)

Not pretty but it'll work.

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**